src/app/layouts/ModalWindow.js

```diff
@@ -0,0 +1,34 @@
+export class ModalWindow {
```

Comment:
Tak jet OK, takie bardziej ćwiczonko - dałbyś radę to zrobić jako funkcja zamiast klasa? 
No bo w sumie nie odwołujesz się nigdzie do this, czyli nawet nie masz stanu - funkcja wystarczy.

Reply:
no właśnie o to pytałem poprzednio -jak wtedy miałbym się odwoływać do funkcji show() i close() ?

Reply:
A to Ci podałem nawet na Discord przykład :) : 
![image](https://user-images.githubusercontent.com/18637116/103912544-ba785000-5107-11eb-87cf-567d4749dcb0.png)
Mogłeś spytać zawsze o doprecyzowanie, wtedy to jest tak:
```
const ModalWindow = () => {
  const modalWindow = document.createElement('div');
  modalWindow.classList.add('modalWindow');
  modalWindow.id = 'modalWindow';

  const parent = document.getElementById('swquiz-app');
  parent.appendChild(modalWindow);
  return {
    show (content) {
      // ""Komponent umożliwia wyświetlenie okna modalnego z zawartością przekazaną jako inny komponent na wejściu do funkcji.""
      // wiadomo jaki będzie typ komponentu jaki przyjdzie? Bo if (content) to chyba średni warunek...
      if (content) {
        content.classList.add('modalWindow__content');

        // create background for content
        const contentBackground = document.createElement('div');
        contentBackground.classList.add('modalWindow__contentBackground');
        contentBackground.id = 'modalWindow-bg';
        modalWindow.appendChild(contentBackground);

        contentBackground.appendChild(content);
        modalWindow.style.display = 'block';
      }
    },
    close () {
      modalWindow.style.display = 'none';

      const contentToRemove = document.getElementById('modalWindow-bg');
      contentToRemove.remove();
    },
  };
};
```

Rozumiesz jak to działa :) ?

Reply:
```
return {
...modalWindow,
show()
}
```

Reply:
@nowakprojects - Wytłumacz mi proszę dlaczego nie można (lub nie powinniśmy) użyć klasy? (one są takie piękne - wręcz mają klasę!)

Reply:
@PiotrWR W TypeScript jak najbardziej.
Tutaj, jako wstep do rozmowy:
https://medium.com/@vapurrmaid/should-you-use-classes-in-javascript-82f3b3df6195
https://www.youtube.com/watch?v=E_BRt_fqaeA&list=PLjHmWifVUNMLR1Oz3WYK7XRLEY5LvSOPu&index=6

Funkcje maja wiecej mozliwosci jak np. zmienne prywatne + mniejszy narzut składni.